### PR TITLE
Retry ABORTED Datastore commit only when NON_TRANSACTIONAL

### LINKED ISSUE #1230: We should not retry ABORTED transactions
In datastore we have `ABORTED` among retryable exceptions. This causes `commit` requests failed due to contention to be retried. Retrying an `ABORTED` transaction always fails with `INVALID ARGUMENT` error.

I suggest we never retry `ABORTED`. \cc @aozarov @ajkannan

### PR DESCRIPTION
**Pull request by mziccard**
Fixes #1230

### CODE REVIEW

**Comment by ajkannan** [comment]
LGTM

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java [hunk 1]
```diff
@@ -38,13 +38,18 @@
   private static final long serialVersionUID = 2663750991205874435L;
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java]: It is unclear to me (@eddavisson may add some clarity) if DEADLINE_EXCEEDED or UNAVAILABLE should be treated as "rejected" suggesting the server did not process the request (the latter sounds more like it than the former).
And if not "rejected" are those still safe to retry the commit (If not, doc should be clear about it).

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java [hunk 2]
```diff
@@ -130,7 +134,7 @@ public CommitResponse commit(CommitRequest request) {
     try {
       return client.commit(request);
     } catch (com.google.datastore.v1.client.DatastoreException ex) {
-      throw translate(ex);
+      throw translate(ex, request.getMode() == CommitRequest.Mode.NON_TRANSACTIONAL);
```
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java]: We could add a retry-logic to [`runInTransaction`](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreHelper.java#L92) as a feature later on.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java (hunk 1 above)
Comment (eddavisson) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreException.java]: If you get DEADLINE_EXCEEDED, you don't know if the underlying commit was triggered or not (or whether it succeeded).

If you get UNAVAILABLE, I believe it is safe to retry the commit (we should only return that before we trigger the underlying commit). UNAVAILABLE can also be returned client-side if the HTTP request fails.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java (hunk 2 above)
Comment (aozarov) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java]: Create an issue?

**Comment by aozarov** [comment]
LGTM

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java (hunk 2 above)
Comment (mziccard) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/DefaultDatastoreRpc.java]: Sure thing, I will create one myself

**Pull request merged**